main/src/ca/uwaterloo/flix/language/phase/jvm/GenFunAndClosureClasses.scala

```diff
@@ -195,6 +218,31 @@ object GenFunAndClosureClasses {
     m.visitEnd()
   }
 
+  private def compileStaticInvokeMethod(visitor: ClassWriter, className: JvmName, defn: Def)(implicit root: Root): Unit = {
+    val m = visitor.visitMethod(ACC_PUBLIC + ACC_FINAL, BackendObjType.Thunk.InvokeMethod.name,
+      AsmOps.getMethodDescriptor(Nil, JvmType.Reference(BackendObjType.Result.jvmName)), null, null)
+    m.visitCode()
+
+    val functionInterface = JvmOps.getFunctionInterfaceType(defn.arrowType).jvmName
+    // Putting args on the Fn class
+    val paramTypes = defn.fparams.map(fp => (fp.tpe, BackendType.toBackendType(fp.tpe)))
+    for (((tpe, erased), i) <- paramTypes.zipWithIndex) {
```

Comment:
this `erased` is not erased. I think its nicer to loop though `(tpe, i)<- defn.fparams.map(_.tpe).zipWithIndex` and then inside compute `val btype = BackendType.toBackendType(tpe)`